### PR TITLE
test(xslt): more tests for run-as=external

### LINKED ISSUE #1010: More tests for transform()-based test
https://github.com/xspec/xspec/pull/892#issue-396249022
> I have a few more tests in my local repository, but I didn't commit them to this branch for fear of cluttering this pull request prematurely. Once we agree on the outlook of this feature, I'll commit them to this branch.

### PR DESCRIPTION
**Pull request by AirQuick**
Closes #1010 

I selected some of the tests in my local repository and packed them into this pull request. (The other tests in my local repository will contribute little to the quality of code.)

### CODE REVIEW

File: test/global-override.xspec [hunk 1]
```diff
@@ -1,47 +1,19 @@
...
-			<x:scenario label="call-template invocation">
-				<x:call template="mirror:context-mirror" />
-				<x:expect label="x:context/@select">
-					<context-grandchild />
-				</x:expect>
-			</x:scenario>
-		</x:scenario>
+	<x:scenario label="Global x:variable in XSpec">
+		<x:call template="get-global-variable" />
+		<x:expect label="overrides xsl:variable in SUT"
```
Comment (galtm) [inline test/global-override.xspec:15]: In a separate PR, do you think it's worthwhile to test that a global `xsl:variable` in the SUT can be overridden by a global `x:param` in the test file? The wiki https://github.com/xspec/xspec/wiki/Writing-Scenarios#global-parameters documents `x:param` as overriding globals in the SUT, whether it's a global parameter or global variable.

Also, this file and `template-context.xspec` aren't related to `run-as=external`, are they? They look like independent refactoring for clarity.

Comment (AirQuick) [reply test/global-override.xspec]: > In a separate PR, do you think it's worthwhile to test that a global `xsl:variable` in the SUT can be overridden by a global `x:param` in the test file?

Done in 550804f35f9d8d8c6227de2aa7d753efedf0e4bc.
(`external_global-override.xspec` and `global-override.xspec` have been consolidated and moved to `test/end-to-end/cases/`. So, done in this PR.)

> Also, this file and `template-context.xspec` aren't related to `run-as=external`, are they? They look like independent refactoring for clarity.

[`external_template-context.xspec`](https://github.com/xspec/xspec/blob/e2de623afe020f777b7f7ca04301f829d10ee899/test/external_template-context.xspec) imports `template-context.xspec`.
[`external_global-override.xspec`](https://github.com/xspec/xspec/blob/39b87c1123bd264711d3da7f1f1f1bdb9055bc2c/test/external_global-override.xspec) was a slightly modified copy of [`global-override.xspec`](https://github.com/xspec/xspec/blob/39b87c1123bd264711d3da7f1f1f1bdb9055bc2c/test/global-override.xspec) (although 550804f35f9d8d8c6227de2aa7d753efedf0e4bc has consolidated them).
That's why this PR contains `template-context.xspec` and `global-override.xspec` along with `run-as=external` files.

Comment (galtm) [reply test/global-override.xspec]: > That's why this PR contains

Thanks. I hadn't looked at the whole set of files and didn't make the connection. This PR looks good to me.

**Review by galtm — APPROVED** [review]
Looks good.

**Pull request merged**